pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -185,4 +136,84 @@ public List<Operator> callJob() {
       return new DistinctCombineOperator(operators, _queryContext, _executorService);
     }
   }
+
+  private List<Operator> getOperatorsSingleThreaded() {
+    List<Operator> operators = new ArrayList<>(_planNodes.size());
+    for (PlanNode planNode : _planNodes) {
+      operators.add(planNode.run());
+    }
+    return operators;
+  }
+
+  private List<Operator> getOperatorsMultiThreaded(int numThreads) {
```

Comment:
this is a little confusing. There is an active scope at the point the thread pool is submitted to, which will be migrated into the thread pool and used as the parent of any traced tasks which take place within the thread pool. We want to record that the current scope delegated to N threads, so we should call `setNumTasks(numThreads)` to record that, and to be consistent with `BaseCombineOperator`.

I don't think we need to trace the tasks completed within the thread pool, because they're not expected to take very long, and they just represent an arbitrary partition of the planning work. What's important is to record that we blocked a thread for Xus here, used N threads, and had M plan nodes.

Reply:
I added this scope because the thread is registered to the trace, but nothing is traceable currently under the thread, so the thread will have empty trace info. But I guess that is okay. Removed the scope here to have the same behavior with the combine operator.

Reply:
I think we just want a single scope to record the time the submitting thread was blocked. The tasks themselves look lightweight and won't make thresholds, but we might even capture some contention on submission to the queue if we move the scope to where the submission is done.

Reply:
the actual implementation ensures that this some would propagate into the thread pool for lineage purposes, so any scope created within the thread pool would have the right parent, if that's what you want to achieve